xla/service/gpu/fusions/transpose_test.cc

```diff
@@ -79,9 +79,13 @@ TEST_F(TransposeTest, ThreadIndexing021) {
       MatchIndexingString(R"(
         (d0, d1, d2, d3, d4, d5)[s0, s1, s2] -> (
           d3 floordiv 2,
-          d0 floordiv 32 + s1 * 4,
-          (d3 mod 2) * 32 + d0 mod 32
-        )
+          d0 floordiv 32 + s1 * 4,)"
+#ifdef GOOGLE_CUDA
+          R"((d3 mod 2) * 32 + d0 mod 32)"
+#elif TENSORFLOW_USE_ROCM
+          R"(d0 mod 32 + (d3 mod 2) * 32)"
+#endif
```

Comment:
Do you understand what's different in ROCm codegen that causes this difference? The expressions are equivalent, so something the construction is not deterministic. I would rather fix that inside indexing maps.